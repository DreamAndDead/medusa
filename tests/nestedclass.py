class Foo:
    class Bar:
        def __init__(self):
            print("__init__ from Bar")
    
    def __init__(self):
        print("__init__ from Foo")
        Foo.Bar()

Foo()
Foo.Bar()