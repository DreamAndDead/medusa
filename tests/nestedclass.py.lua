dofile('./pythonlua/luainit.lua')

local Foo = class(function(Foo)
    Foo.Bar = class(function(Bar)
        function Bar.__init__(self)
            print("__init__ from Bar")
        end
        return Bar
    end, {})
    function Foo.__init__(self)
        print("__init__ from Foo")
        Foo.Bar()
    end
    return Foo
end, {})
Foo()
Foo.Bar()