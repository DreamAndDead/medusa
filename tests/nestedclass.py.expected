__init__ from Foo
__init__ from Bar
__init__ from Bar
